docs/advanced/sts.md

```diff
@@ -0,0 +1,60 @@
+### Strict transport security
+
+Due to recent development efforts, KICL now supports the IRCv3
+[STS specification](https://ircv3.net/specs/core/sts-3.3.html) which adds support for strict transport security
+policies to be set by the IRC server administrators.
+
+These policies tell the client to connect **only via a secure TLS connection** for a given duration. They also specify
+the port on which the client can reach the secure service. Further details on how the system works can be found in the
+[GitHub issue](https://github.com/KittehOrg/KittehIRCClientLib/issues/139) and in the aforementioned specification.
+
+Currently, STS support in KICL is opt-in but in the future this may change to give new developers protection by default
+whilst ensuring experienced developers are able to turn off the functionality.
+
+#### Why should I care?
+
+The internet is a scary place now. We can't afford to do things in plaintext anymore, and IRC is no exception.
```

Comment:
The first sentence isn't entirely serious with its wording (was copied from the original issue) but I stand by the sentiment expressed. it is irresponsible to create and provide servers for internet protocols that only operate in plaintext. STS is one way we can try to migrate to an encrypted-only IRC.

How would you suggest wording this?